tests/test_gpu_treeshap.cu

```diff
@@ -288,41 +349,43 @@ TEST(GPUTreeShap, BasicPathsInteractions) {
       {2, 1, 0, -inf, 0.5f, false, 1.0f / 3, -1.0f},
       {3, -1, 0, -inf, 0.0f, false, 1.0f, -1.0f},
       {3, 0, 0, -inf, 0.5f, false, 0.4f, -1.0f}};
-  thrust::device_vector<float> data = std::vector<float>({1.0f, 1.0f, 0.0f});
-  DenseDatasetWrapper X(data.data().get(), 1, 3);
+  thrust::device_vector<float> data =
+      std::vector<float>({1.0f, 1.0f, 0.0f, 1.0f, 1.0f, 1.0f});
+  DenseDatasetWrapper X(data.data().get(), 2, 3);
   thrust::device_vector<float> phis(X.NumRows() * (X.NumCols() + 1) *
                                     (X.NumCols() + 1));
-  GPUTreeShapInteractions(X, path.begin(), path.end(), 1, phis.data().get(), phis.size());
-  thrust::host_vector<float> result(phis);
-  // First instance
-  // EXPECT_NEAR(result[0], 0.6277778f * num_trees, 1e-5);
-  // EXPECT_NEAR(result[1], 0.5027776f * num_trees, 1e-5);
-  // EXPECT_NEAR(result[2], 0.1694444f * num_trees, 1e-5);
-  // EXPECT_NEAR(result[3], -0.3f * num_trees, 1e-5);
-  //// Second instance
-  // EXPECT_NEAR(result[4], 0.24444449f * num_trees, 1e-5);
-  // EXPECT_NEAR(result[5], -1.005555f * num_trees, 1e-5);
-  // EXPECT_NEAR(result[6], 0.0611111f * num_trees, 1e-5);
-  // EXPECT_NEAR(result[7], -0.3f * num_trees, 1e-5);
+  GPUTreeShapInteractions(X, path.begin(), path.end(), 1, phis.data().get(),
+                          phis.size());
+  std::vector<float> result(phis.begin(), phis.end());
+  std::vector<float> expected_result = {
```

Comment:
How did you come up with these?

Reply:
It's from a simple xgboost tree, I added a comment showing the tree structure.